cmd/stellar-rpc/internal/methods/get_ledger_entries.go

```diff
@@ -19,8 +22,112 @@ var ErrLedgerTTLEntriesCannotBeQueriedDirectly = "ledger ttl entries cannot be q
 
 const getLedgerEntriesMaxKeys = 200
 
-// NewGetLedgerEntriesHandler returns a JSON RPC handler to retrieve the specified ledger entries from Stellar Core.
-func NewGetLedgerEntriesHandler(logger *log.Entry, ledgerEntryReader db.LedgerEntryReader) jrpc2.Handler {
+type ledgerEntryGetter interface {
+	GetLedgerEntries(ctx context.Context, keys []xdr.LedgerKey) ([]db.LedgerKeyAndEntry, uint32, error)
+}
+
+type coreLedgerEntryGetter struct {
+	coreClient         interfaces.FastCoreClient
+	latestLedgerReader db.LedgerEntryReader
+}
+
+func (c coreLedgerEntryGetter) GetLedgerEntries(
+	ctx context.Context,
+	keys []xdr.LedgerKey,
+) ([]db.LedgerKeyAndEntry, uint32, error) {
+	latestLedger, err := c.latestLedgerReader.GetLatestLedgerSequence(ctx)
+	if err != nil {
+		return nil, 0, fmt.Errorf("could not get latest ledger: %w", err)
+	}
+	// Pass latest ledger here in case Core is ahead of us (0 would be Core's latest).
+	resp, err := c.coreClient.GetLedgerEntries(ctx, latestLedger, keys...)
+	if err != nil {
+		return nil, 0, fmt.Errorf("could not query captive core: %w", err)
+	}
```

Comment:
@SirTyson just wanna double-check that this logic makes sense: we want Core to use the latest ledger that RPC is aware of

Reply:
Yep, looks good!